modules/metrics/dropwizard/src/test/scala/metricsops/DropWizardMetricsOpsTests.scala

```diff
@@ -77,88 +77,74 @@ object DropWizardMetricsOpsTests extends Properties("DropWizardMetrics") {
     Status.UNKNOWN
   )
 
-  def methodTypeName(methodType: MethodType): String = methodType match {
-    case MethodType.UNARY            => "unary"
-    case MethodType.CLIENT_STREAMING => "client-streaming"
-    case MethodType.SERVER_STREAMING => "server-streaming"
-    case MethodType.BIDI_STREAMING   => "bidi-streaming"
-    case MethodType.UNKNOWN          => "unknown"
-  }
-
   def performAndCheckMetrics(
       methodInfo: GrpcMethodInfo,
       numberOfCalls: Int,
       expectedCount: Int,
       registry: MetricRegistry,
       gaugeName: String,
       gaugeType: GaugeType,
-      f: () => IO[Unit],
-      status: Option[Status] = None): Boolean = {
-
+      op: IO[Unit],
+      status: Option[Status] = None): IO[Boolean] =
     (1 to numberOfCalls).toList
-      .map { _ =>
-        f()
-      }
-      .sequence
-      .unsafeRunSync()
-
-    checkMetrics(expectedCount, registry, gaugeName, gaugeType)
-  }
+      .map(_ => op)
+      .sequence_
+      .map(_ => checkMetrics(expectedCount, registry, gaugeName, gaugeType))
 
   def checkMetrics(
       expectedCount: Int,
       registry: MetricRegistry,
       gaugeName: String,
       gaugeType: GaugeType
-  ): Boolean =
-    gaugeType match {
-      case Timer =>
-        registry.getTimers().asScala.contains(gaugeName) && registry
-          .timer(gaugeName)
-          .getCount == expectedCount
-      case Counter =>
-        registry
-          .getCounters()
-          .asScala
-          .contains(gaugeName) && registry
-          .counter(gaugeName)
-          .getCount == expectedCount
-      case _ => false
-    }
+  ): Boolean = {
+    val (gaugeMap, gaugeCount) =
+      gaugeType match {
+        case Timer =>
+          (registry.getTimers().asScala.toMap, registry.timer(gaugeName).getCount)
+        case Counter =>
+          (registry.getCounters().asScala.toMap, registry.counter(gaugeName).getCount)
+        case _ =>
+          (Map.empty[String, GaugeType], 0)
+      }
+    gaugeMap.contains(gaugeName) && gaugeCount == expectedCount
+  }
 
   property("creates and updates counter when registering an active call") =
-    forAll(methodInfoGen, Gen.chooseNum[Int](1, 10)) {
+    forAllNoShrink(methodInfoGen, Gen.chooseNum[Int](1, 10)) {
```

Comment:
```suggestion
    forAll(methodInfoGen, Gen.chooseNum[Int](1, 10)) {
```

Reply:
I'm inclined to use `forAllNoShrink` because if it fails, it starts to try with empty method names, negative calls and things like that. I think the `forAll` is good for other scenarios, wdyt?